README.md

```diff
@@ -1,9 +1,109 @@
+
+[See the input repository](https://github.com/minimization/content-resolver-input)
+
+### Main concepts
+
+![Workloads and Environments](docs/img/workloads-and-envs.png)
+
+*Workloads* are the primary reason for Content Resolver's existence, representing a package set with a specific purpose — an application, a runtime, a set of dependencies for a specific se case, etc.
+
+Workloads consist of two parts:
+* Required Packages — defined by the user
+* Dependencies — resolved by Content Resolver
+
+Workloads are resolved on top of *Environments* which are meant to represent environments in which workloads would typically run — container base images, cloud images, or even desktop installations.
+
+Similar to workloads, environments consist of the same two parts:
+* Required Packages — defined by the user
+* Dependencies — resolved by Content Resolver
+
+Environments are fully resolved first, and workloads are then resolved on top of them.
+
+*Repositories* are the sources of packages Content Resolver uses to resolve Workloads and Environments.
+
+Finally, everything is connected b *Labels*. Workloads are resolved on top of environments with the same label, using repositories with (you guessed it!) the same label.
+
+### Views
+
+![Views](docs/img/views.png)
+
+*Views* are an advanced concept that combines multiple workloads into a single view.
+
+Views can serve many purposes, but the primary reason for their existence is [Fedora ELN](https://docs.fedoraproject.org/en-US/eln/). Multiple contributors are able to define a subset of Fedora Rawhide to be rebuilt with Enterprise Linux flags.
+
+For this, Content Resolver can do additional things with Views:
+* resolve build dependencies
+* show detailed reasons of presence of every package
+* recommend maintainers of shared dependencies
+* mark and track unwanted packages
+
+### Build Dependencies (draft)
+
+Content Resolver can get build dependencies resolved for views, showing what needs to be in the buildroot in order to rebuild the entire view, including all the build dependencies themselves.
+
+This way, the entire set (required packages + runtime dependencies + build dependencies) is self-hosting.
+
+Build dependencies are currently resolved in an external service (dep-tracker) because every SRPM (source package) needs to be rebuilt for all architectures. That's necessary because dependencies can vary across different architectures. And unlike with RPMs (binary packages), SRPMs are not distributed per-architecture, as their primary use case is to distribute the sources rather than provide dependency information.
+
+Current limitations: Because the build dependencies are resolved by an external service, the results might be lagging even several hours. There's also currently no distinction between direct build dependencies and dependencies of build dependencies in on the package details page.
```

Comment:
```suggestion
Current limitations: Because the build dependencies are resolved by an external service, the results might lag as much as several hours behind the runtime views. There is also currently no distinction between direct build dependencies and dependencies of build dependencies in on the package details page.
```